packages/gensx-cli/src/commands/login.ts

```diff
@@ -0,0 +1,216 @@
+    "base64url",
+  );
+}
+
+function createCodeHash(code: string): string {
+  return createHash("sha256").update(code).digest("base64url");
+}
+
+async function saveConfig(config: Config): Promise<void> {
+  const { configDir, configFile } = getConfigPath();
+
+  try {
+    await mkdir(configDir, { recursive: true, mode: 0o700 });
+
+    const configContent = stringifyIni({
+      api: {
+        token: config.token,
+        org: config.orgSlug,
+        baseUrl: API_BASE_URL,
+      },
+    });
+
+    // Add a helpful header comment
+    const fileContent = `; GenSX Configuration File
+; Generated on: ${new Date().toISOString()}
+
+${configContent}`;
+
+    const mode = platform() === "win32" ? undefined : 0o600;
+    await writeFile(configFile, fileContent, { mode });
+  } catch (err) {
+    const message =
+      err instanceof Error ? err.message : "Unknown error occurred";
+    throw new Error(`Failed to save configuration: ${message}`);
+  }
+}
+
+async function createLoginRequest(
+  verificationCode: string,
+): Promise<DeviceAuthRequest> {
+  const url = new URL("/auth/device/request", API_BASE_URL);
+  const response = await fetch(url, {
+    method: "POST",
+    headers: {
+      "Content-Type": "application/json",
+    },
+    body: JSON.stringify({
+      clientId: hostname(),
+      codeChallenge: createCodeHash(verificationCode),
+      codeChallengeMethod: "S256",
+    }),
+  });
+
+  if (!response.ok) {
+    throw new Error(`Failed to create login request: ${response.statusText}`);
+  }
+
+  const body = (await response.json()) as {
+    status: "ok";
+    data: DeviceAuthRequest;
+  };
+  if (!body.data.requestId || !body.data.expiresAt) {
+    throw new Error("Invalid response from server");
+  }
+
+  return body.data;
+}
+
+async function pollLoginStatus(
+  requestId: string,
+  verificationCode: string,
+): Promise<DeviceAuthStatus> {
+  const url = new URL(`/auth/device/request/${requestId}`, API_BASE_URL);
+  url.searchParams.set("code_verifier", verificationCode);
+  const response = await fetch(url);
+
+  if (!response.ok) {
+    throw new Error(`Failed to check login status: ${response.statusText}`);
+  }
+
+  const body = (await response.json()) as {
+    status: "ok";
+    data: DeviceAuthStatus;
+  };
+  if (body.data.status === "pending") {
+    return { status: "pending" };
+  }
+
+  if (body.data.status === "expired") {
+    throw new Error("Login expired");
+  }
+
+  return body.data;
+}
+
+export async function login(): Promise<void> {
+  const spinner = ora();
+
+  try {
+    spinner.start("Logging in to GenSX");
+
+    const verificationCode = generateVerificationCode();
+    const request = await createLoginRequest(verificationCode);
+    spinner.succeed();
+
+    const authUrl = new URL(`/auth/device/${request.requestId}`, APP_BASE_URL);
+    authUrl.searchParams.set("code_verifier", verificationCode);
+
+    consola.box(
+      `Press any key to open your browser and authenticate with GenSX:
```

Comment:
Let's go ahead and ship this, but we should probably add a fallback incase this fails that let's users paste in an access code. Maybe open a follow up issue?

Reply:
Good point. It does print the URL, but it is not polling at that point.

Reply:
Unfortunately we need the API key _and_ org....

Reply:
Once we have an API key, we can poll for available orgs. 

If there is already an org in the config file, we could just skip writing one.